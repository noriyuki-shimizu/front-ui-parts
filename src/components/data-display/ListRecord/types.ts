import type { AnchorHTMLAttributes } from 'vue'

/** レコード全体をリンクで囲う際のリンクデータ */
export interface LinkBridge extends AnchorHTMLAttributes {
  ariaLabel?: string
}

/** Props */
export interface Props {
  /** レコード全体をリンクで囲う際のリンクデータ */
  linkBridge?: LinkBridge
}
