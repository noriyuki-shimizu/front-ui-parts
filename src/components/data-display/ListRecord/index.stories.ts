import type { ArgTypeObject, Meta, StoryObj } from '@storybook/vue3'
import ListRecord from './index.vue'

const meta = {
  title: 'Data Display/ListRecord',
  // TODO: Watch: https://github.com/storybookjs/storybook/issues/24238
  component: ListRecord as unknown as Record<string, unknown>,
  tags: ['autodocs'],
  argTypes: {
    /* slots.img */
    img: {
      description: '画像',
      type: 'img',
      control: 'object'
    },
    /* slots.leftAside */
    leftAside: {
      description: '左アサイドへの表示項目',
      type: 'string',
      control: 'text'
    },
    /* slots.header */
    header: {
      description: 'ヘッダー',
      type: 'string',
      control: 'text'
    },
    /* slots.title */
    title: {
      description: 'タイトル',
      type: 'string',
      control: 'text',
      table: {
        defaultValue: {
          summary: 'タイトル'
        }
      }
    },
    /* slots.description */
    description: {
      description: '説明',
      type: 'string',
      control: 'text'
    },
    /* slots.footer */
    footer: {
      description: 'フッター',
      type: 'string',
      control: 'text'
    }
  }
} satisfies Meta<typeof ListRecord> & ArgTypeObject // TODO: Watch: https://github.com/storybookjs/storybook/issues/23352

export default meta

type Story = StoryObj<typeof meta>

export const SimpleRecord: Story = {
  render: (args) => ({
    components: { ListRecord: ListRecord as unknown as Record<string, unknown> },
    setup() {
      return { args }
    },
    template: `
    <ListRecord>
      <template #header>ヘッダー</template>
      <template #title>タイトル</template>
      <template #description>ここに説明文を記述します。</template>
      <template #footer>フッター</template>
    </ListRecord>
    `
  })
}

export const LeftAsideRecord: Story = {
  render: (args) => ({
    components: { ListRecord: ListRecord as unknown as Record<string, unknown> },
    setup() {
      return { args }
    },
    template: `
    <ListRecord>
      <template #leftAside>
        <span style="font-size: 40px;">😄</span>
      </template>
      <template #header>ヘッダー</template>
      <template #title>タイトル</template>
      <template #description>ここに説明文を記述します。</template>
      <template #footer>フッター</template>
    </ListRecord>
    `
  })
}

export const ImgRecord: Story = {
  render: (args) => ({
    components: { ListRecord: ListRecord as unknown as Record<string, unknown> },
    setup() {
      return { args }
    },
    template: `
    <ListRecord>
      <template #img>
        <img
          src="https://github.com/noriyuki-shimizu/front-ui-parts/blob/main/src/assets/img/nature.jpg?raw=true"
          alt="nature"
          width="100%"
          height="100%"
        />
      </template>
      <template #header>ヘッダー</template>
      <template #title>タイトル</template>
      <template #description>ここに説明文を記述します。</template>
      <template #footer>フッター</template>
    </ListRecord>
    `
  })
}

export const LongTextRecord: Story = {
  render: (args) => ({
    components: { ListRecord: ListRecord as unknown as Record<string, unknown> },
    setup() {
      return { args }
    },
    template: `
    <ListRecord>
      <template #header>
        長い文字列のヘッダー長い文字列のヘッダー長い文字列のヘッダー長い文字列のヘッダー長い文字列のヘッダー長い文字列のヘッダー長い文字列のヘッダー長い文字列のヘッダー長い文字列のヘッダー長い文字列のヘッダー
      </template>
      <template #title>
        長い文字列のタイトル長い文字列のタイトル長い文字列のタイトル長い文字列のタイトル長い文字列のタイトル長い文字列のタイトル長い文字列のタイトル長い文字列のタイトル長い文字列のタイトル長い文字列のタイトル
      </template>
      <template #description>
        長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文。長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文。
      </template>
      <template #footer>
        長い文字列のフッター長い文字列のフッター長い文字列のフッター長い文字列のフッター長い文字列のフッター長い文字列のフッター長い文字列のフッター長い文字列のフッター長い文字列のフッター長い文字列のフッター
      </template>
    </ListRecord>
    `
  })
}
