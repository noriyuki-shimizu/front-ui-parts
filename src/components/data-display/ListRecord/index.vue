<script setup lang="ts">
import { LangUtil } from '@/utils/core'
</script>

<template>
  <div :class="$style['list-detail']">
    <div v-if="!LangUtil.isUndefined($slots.img)" :class="$style['list-detail__img-wrapper']">
      <slot name="img" />
    </div>
    <div
      :class="[
        $style['list-detail__body'],
        $style['body'],
        { [$style['body--shift-right']]: !LangUtil.isUndefined($slots.img) }
      ]"
    >
      <div v-if="!LangUtil.isUndefined($slots.leftAside)" :class="$style['body__left-aside']">
        <slot name="leftAside" />
      </div>
      <article :class="$style['body__container']">
        <header v-if="!LangUtil.isUndefined($slots.header)" :class="$style['body__header']">
          <slot name="header" />
        </header>
        <h2 :class="$style['body__title']">
          <slot name="title">タイトル</slot>
        </h2>
        <div v-if="!LangUtil.isUndefined($slots.description)" :class="$style['body__description-wrapper']">
          <p :class="$style['body__description']">
            <slot name="description" />
          </p>
        </div>
        <footer v-if="!LangUtil.isUndefined($slots.footer)" :class="$style['body__footer']">
          <slot name="footer" />
        </footer>
      </article>
    </div>
  </div>
</template>

<style module lang="scss">
@use './style.scss';
</style>
