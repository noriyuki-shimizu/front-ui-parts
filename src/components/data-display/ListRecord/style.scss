.list-detail {
  position: relative;
  border: 1px solid var(--color-border-normal);
  border-radius: 8px;

  &__img-wrapper {
    position: absolute;
    left: 0;
    width: 240px;
    height: 100%;
    object-fit: cover;
    border-radius: 8px 0 0 8px;
  }
}

.body {
  display: flex;
  padding: 16px 16px 16px 0;

  &--shift-right {
    padding-left: 240px;
  }

  &__left-aside {
    max-width: 80px;
    padding-left: 16px;
    margin: auto 0;
  }

  &__container {
    padding-left: 16px;
  }

  &__header {
    display: flex;
    align-items: center;
    margin-bottom: 8px;
  }

  &__title {
    margin-bottom: 8px;
    font-size: 24px;
    line-height: 1.5;
  }

  &__description-wrapper {
    width: 100%;
    margin-bottom: 8px;
    overflow: hidden;
  }

  &__description {
    display: -webkit-box;
    font-size: 16px;
    line-height: 1.5;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 2;
  }

  &__footer {
    font-size: 12px;
  }
}
