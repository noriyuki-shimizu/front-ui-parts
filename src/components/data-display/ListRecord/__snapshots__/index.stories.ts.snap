// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Display/ListRecord ImgRecord smoke-test 1`] = `
<div class="_list-detail_9np8i_325">
  <div class="_list-detail__img-wrapper_9np8i_330">
    <img src="https://github.com/noriyuki-shimizu/front-ui-parts/blob/main/src/assets/img/nature.jpg?raw=true"
         alt="nature"
         width="100%"
         height="100%"
    >
  </div>
  <div class="_body_9np8i_339 _body--shift-right_9np8i_343">
    <article class="_body__container_9np8i_351">
      <header class="_body__header_9np8i_354">
        ヘッダー
      </header>
      <h2 class="_body__title_9np8i_359">
        タイトル
      </h2>
      <div class="_body__description-wrapper_9np8i_364">
        <p class="_body__description_9np8i_364">
          ここに説明文を記述します。
        </p>
      </div>
      <footer class="_body__footer_9np8i_376">
        フッター
      </footer>
    </article>
  </div>
</div>
`;

exports[`Data Display/ListRecord LeftAsideRecord smoke-test 1`] = `
<div class="_list-detail_9np8i_325">
  <div class="_body_9np8i_339">
    <div class="_body__left-aside_9np8i_346">
      <span style="font-size: 40px;">
        😄
      </span>
    </div>
    <article class="_body__container_9np8i_351">
      <header class="_body__header_9np8i_354">
        ヘッダー
      </header>
      <h2 class="_body__title_9np8i_359">
        タイトル
      </h2>
      <div class="_body__description-wrapper_9np8i_364">
        <p class="_body__description_9np8i_364">
          ここに説明文を記述します。
        </p>
      </div>
      <footer class="_body__footer_9np8i_376">
        フッター
      </footer>
    </article>
  </div>
</div>
`;

exports[`Data Display/ListRecord LongTextRecord smoke-test 1`] = `
<div class="_list-detail_9np8i_325">
  <div class="_body_9np8i_339">
    <article class="_body__container_9np8i_351">
      <header class="_body__header_9np8i_354">
        長い文字列のヘッダー長い文字列のヘッダー長い文字列のヘッダー長い文字列のヘッダー長い文字列のヘッダー長い文字列のヘッダー長い文字列のヘッダー長い文字列のヘッダー長い文字列のヘッダー長い文字列のヘッダー
      </header>
      <h2 class="_body__title_9np8i_359">
        長い文字列のタイトル長い文字列のタイトル長い文字列のタイトル長い文字列のタイトル長い文字列のタイトル長い文字列のタイトル長い文字列のタイトル長い文字列のタイトル長い文字列のタイトル長い文字列のタイトル
      </h2>
      <div class="_body__description-wrapper_9np8i_364">
        <p class="_body__description_9np8i_364">
          長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文。長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文長い文字列の説明文。
        </p>
      </div>
      <footer class="_body__footer_9np8i_376">
        長い文字列のフッター長い文字列のフッター長い文字列のフッター長い文字列のフッター長い文字列のフッター長い文字列のフッター長い文字列のフッター長い文字列のフッター長い文字列のフッター長い文字列のフッター
      </footer>
    </article>
  </div>
</div>
`;

exports[`Data Display/ListRecord SimpleRecord smoke-test 1`] = `
<div class="_list-detail_9np8i_325">
  <div class="_body_9np8i_339">
    <article class="_body__container_9np8i_351">
      <header class="_body__header_9np8i_354">
        ヘッダー
      </header>
      <h2 class="_body__title_9np8i_359">
        タイトル
      </h2>
      <div class="_body__description-wrapper_9np8i_364">
        <p class="_body__description_9np8i_364">
          ここに説明文を記述します。
        </p>
      </div>
      <footer class="_body__footer_9np8i_376">
        フッター
      </footer>
    </article>
  </div>
</div>
`;
